src/overlays/actors/ovl_En_Mushi2/z_en_mushi2.c

```diff
@@ -30,134 +45,1190 @@ const ActorInit En_Mushi2_InitVars = {
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Mushi2/func_80A68BC8.s")
+    Actor_SetVelocityAndMoveYRotationAndGravity(&this->actor);
+    func_80A69424(this, globalCtx);
+    Math_StepToF(&this->actor.world.pos.y, this->actor.world.pos.y + this->actor.yDistToWater, 2.8f);
+    this->skelAnime.animPlaybackSpeed = this->unk_368 * 0.018f;
+    this->skelAnime.animPlaybackSpeed = CLAMP(this->skelAnime.animPlaybackSpeed, 0.1f, 1.9f);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Mushi2/func_80A68C5C.s")
+    if (this->unk_368 > 80) {
+        this->unk_36E += Rand_S16Offset(-50, 100);
+        this->unk_370 += Rand_S16Offset(-300, 600);
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Mushi2/func_80A68CE4.s")
+    temp = this->skelAnime.animPlaybackSpeed * 200.0f;
+    this->unk_36E = CLAMP(this->unk_36E, -temp, temp);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Mushi2/func_80A68DD4.s")
+    temp = this->skelAnime.animPlaybackSpeed * 1000.0f;
+    this->unk_370 = CLAMP(this->unk_370, -temp, temp);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Mushi2/func_80A68ED8.s")
+    this->actor.world.rot.y += this->unk_36E;
+    this->actor.shape.rot.y += this->unk_370;
+    Math_ScaledStepToS(&this->actor.world.rot.x, 0, 0xBB8);
+    this->actor.shape.rot.x = this->actor.world.rot.x;
+    Math_ScaledStepToS(&this->actor.world.rot.z, 0, 0xBB8);
+    this->actor.shape.rot.z = this->actor.world.rot.z;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Mushi2/func_80A68F24.s")
+    if ((this->actor.flags & 0x40) && (Rand_ZeroOne() < 0.03f)) {
+        Vec3f sp3C;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Mushi2/func_80A68F9C.s")
+        sp3C.x = this->actor.world.pos.x;
+        sp3C.y = this->actor.world.pos.y + this->actor.yDistToWater;
+        sp3C.z = this->actor.world.pos.z;
+        EffectSsGRipple_Spawn(globalCtx, &sp3C, 40, 200, 4);
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Mushi2/func_80A690C4.s")
+    if ((this->unk_368 <= 0) || (this->unk_36A <= 0) || func_80A68BA0(this)) {
+        func_80A6A094(this);
+        func_80A6A0D8(this);
+        func_80A6AE14(this);
+    } else if (!(this->actor.bgCheckFlags & 0x20)) {
+        func_80A6A794(this);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Mushi2/func_80A691EC.s")
+void func_80A6AE14(EnMushi2* this) {
+    this->unk_30C &= ~1;
+    this->unk_368 = 100;
+    this->actor.velocity.y = 0.0f;
+    this->actor.minVelocityY = -0.8f;
+    this->actor.gravity = -0.04f;
+    func_80A68B6C(this);
+    this->actionFunc = func_80A6AE7C;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Mushi2/func_80A69388.s")
+void func_80A6AE7C(EnMushi2* this, GlobalContext* globalCtx) {
+    f32 temp_f2;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Mushi2/func_80A69424.s")
+    this->actor.shape.rot.x -= 0x1F4;
+    this->actor.shape.rot.y += 200;
```

Comment:
```suggestion
    this->actor.shape.rot.x -= 0x1F4;
    this->actor.shape.rot.y += 0xC8;
```
I've been a lot less picky on dec vs hex angles, but when they're right next to each other like this, I think they should be the same